src/ObservableOption.ts

```diff
@@ -0,0 +1,434 @@
+  <A>(predicate: Predicate<A>): (ma: ObservableOption<A>) => ObservableOption<A>
+} = <A>(predicate: Predicate<A>): ((ma: ObservableOption<A>) => ObservableOption<A>) =>
+  chain(a => (predicate(a) ? of(a) : none))
+
+/**
+ * Derivable from `MonadThrow`.
+ *
+ * @since 0.6.14
+ */
+export const fromOption = <A>(ma: O.Option<A>): ObservableOption<A> => (ma._tag === 'None' ? none : of(ma.value))
+
+/**
+ * Derivable from `MonadThrow`.
+ *
+ * @since 0.6.14
+ */
+export const fromPredicate: {
+  <A, B extends A>(refinement: Refinement<A, B>): (a: A) => ObservableOption<B>
+  <A>(predicate: Predicate<A>): (a: A) => ObservableOption<A>
+} = <A>(predicate: Predicate<A>) => (a: A): ObservableOption<A> => (predicate(a) ? of(a) : none)
+
+// -------------------------------------------------------------------------------------
+// instances
+// -------------------------------------------------------------------------------------
+
+/* istanbul ignore next */
+const map_: Functor1<URI>['map'] = (fa, f) => pipe(fa, map(f))
+/* istanbul ignore next */
+const ap_: Apply1<URI>['ap'] = (fab, fa) => pipe(fab, ap(fa))
+/* istanbul ignore next */
+const chain_: Monad1<URI>['chain'] = (ma, f) => pipe(ma, chain(f))
+/* istanbul ignore next */
+const alt_: Alt1<URI>['alt'] = (fx, fy) => pipe(fx, alt(fy))
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const URI = 'ObservableOption'
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export type URI = typeof URI
+
+declare module 'fp-ts/lib/HKT' {
+  interface URItoKind<A> {
+    readonly [URI]: ObservableOption<A>
+  }
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const Functor: Functor1<URI> = {
+  URI,
+  map: map_
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const Apply: Apply1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const Applicative: Applicative1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_,
+  of
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const Monad: Monad1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_,
+  of,
+  chain: chain_
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const Alt: Alt1<URI> = {
+  URI,
+  map: map_,
+  alt: alt_
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const MonadIO: MonadIO1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_,
+  of,
+  chain: chain_,
+  fromIO
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const MonadTask: MonadTask1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_,
+  of,
+  chain: chain_,
+  fromIO,
+  fromTask
+}
+
+/**
+ * @category instances
+ * @since 0.6.14
+ */
+export const MonadObservable: MonadObservable1<URI> = {
+  URI,
+  map: map_,
+  ap: ap_,
+  of,
+  chain: chain_,
+  fromIO,
+  fromTask,
+  fromObservable
+}
+
+/**
+ * @category instances
+ * @deprecated
+ * @since 0.6.14
+ */
+export const observableOption: Monad1<URI> & Alt1<URI> & MonadObservable1<URI> = {
```

Comment:
we can remove this I guess

Reply:
I added it just because we have `observableEither` in `ObservableEither`. Should we remove that as well?

Reply:
`observableEither` was added in v0.6.8 and now is deprecated precisely because we can't remove it as it would be a breaking change. This is a new module though so there is little point to add a deprecated API

Reply:
b0aa512cea52e14992675a5df1523f0ec7f38d6f